packages/web-components/fast-foundation/src/select/select.spec.ts

```diff
@@ -88,6 +89,279 @@ describe("Select", () => {
+            ]);
+
+            expect(wasChanged).to.be.true;
+
+            expect(element.value).to.equal("two");
+
+            await disconnect();
+        });
+
+        it("via arrow up key", async () => {
+            const { element, connect, disconnect } = await setup();
+
+            await connect();
+
+            element.value = "two";
+
+            expect(element.value).to.equal("two");
+
+            expect(element.open).to.be.false;
+
+            const event = new KeyboardEvent("keydown", {
+                key: "ArrowUp",
+                keyCode: KeyCodes.arrowUp,
+            } as KeyboardEventInit);
+
+            const wasChanged = await Promise.race([
+                new Promise(resolve => {
+                    element.addEventListener("change", () => resolve(true));
+                    element.dispatchEvent(event);
+                }),
+                DOM.nextUpdate().then(() => false),
+            ]);
+
+            expect(wasChanged).to.be.true;
+
+            expect(element.value).to.equal("one");
+
+            await disconnect();
+        });
+
+        it("via home key", async () => {
+            const { element, connect, disconnect } = await setup();
+
+            await connect();
+
+            element.value = "three";
+
+            expect(element.value).to.equal("three");
+
+            expect(element.open).to.be.false;
+
+            const event = new KeyboardEvent("keydown", {
+                key: "Home",
+                keyCode: KeyCodes.home,
+            } as KeyboardEventInit);
+
+            const wasChanged = await Promise.race([
+                new Promise(resolve => {
+                    element.addEventListener("change", () => resolve(true));
+                    element.dispatchEvent(event);
+                }),
+                DOM.nextUpdate().then(() => false),
+            ]);
+
+            expect(wasChanged).to.be.true;
+
+            expect(element.value).to.equal("one");
+
+            await disconnect();
+        });
+
+        it("via end key", async () => {
+            const { element, connect, disconnect } = await setup();
+
+            await connect();
+
+            expect(element.open).to.be.false;
+
+            const event = new KeyboardEvent("keydown", {
+                key: "End",
+                keyCode: KeyCodes.end,
+            } as KeyboardEventInit);
+
+            const wasChanged = await Promise.race([
+                new Promise(resolve => {
+                    element.addEventListener("change", () => resolve(true));
+                    element.dispatchEvent(event);
+                }),
+                DOM.nextUpdate().then(() => false),
+            ]);
+
+            expect(wasChanged).to.be.true;
+
+            expect(element.value).to.equal("three");
+
+            await disconnect();
+        });
+    });
+
+    describe("should NOT emit a 'change' event when the value changes by programmatic interaction", () => {
```

Comment:
For my own confirmation, this is the way the built-ins work as well, yes?

Reply:
Correct - if you have a `<select>` and you change its selected value with `select.value = "any sort of value"`, it won't emit an event.